WorkloadManagementSystem/Utilities/RabbitMQSynchronizer.py

```diff
@@ -0,0 +1,103 @@
+  RabbitMQ user database is updated according to changes in CS.
+'''
+
+from DIRAC import S_OK
+from DIRAC.ConfigurationSystem.Client.Helpers.Registry import hostHasProperties
+from DIRAC.ConfigurationSystem.Client.Helpers.Registry import getDNsInGroup
+from DIRAC.ConfigurationSystem.Client.Helpers.Registry import getDNForHost
+from DIRAC.ConfigurationSystem.Client.Helpers.Registry import getHosts
+from DIRAC.Core.Utilities.RabbitMQAdmin import getAllUsers, deleteUsers
+from DIRAC.Core.Utilities.RabbitMQAdmin import setUsersPermissions, addUsersWithoutPasswords
+from DIRAC.ResourceStatusSystem.Utilities import CSHelpers
+
+class RabbitMQSynchronizer(object):
+
+  def __init__( self ):
+
+    # Warm up local CS
+    # I am not sure whether it is needed but
+    # it was used in DIRAC.ResourceStatusSystem.Utilities.Synchronizer
+    CSHelpers.warmUp()
+    self._accessUserGroup = 'lhcb_pilot'  #only users belonging to group with this property are allowed to connect
+    self._accessProperty = 'GenericPilot' #only host with this property are allowed to connect
+
+  def sync( self, _eventName, _params ):
+    '''Synchronizes the internal RabbitMQ user database with the current content of CS.
+
+    Args:
+      _eventName: any value, this parameter is ignored, but needed by caller function.
+      _params: - any value, this parameter is ignored, but needed by caller function.
+    Returns:
+      S_OK:
+    Example:
+       s.sync( None, None )
+    '''
+
+    valid_users = getDNsInGroup(self._accessUserGroup)
+    valid_hosts = getDNsForValidHosts(self._accessProperty)
+    updateRabbitMQDatabase(valid_users + valid_hosts)
+    return S_OK()
+
+def getDNsForValidHosts(accessProperty):
+  """Returns DN of hosts which contains accessProperty based on current
+     CS settings.
+  Args:
+    accessProperty(str):
+  Returns:
+    list: of hosts with accessProperty set.
+  """
+
+  retVal = getHosts()
+  if not retVal[ 'OK' ]:
+    return []
+  hosts = retVal['Value']
+  DNs = []
+  for host in hosts:
+    if hostHasProperties(host, [accessProperty]):
+      retVal = getDNForHost(host)
+      if retVal[ 'OK' ]:
+        DNs.extend(retVal['Value'])
+      else:
+        print 'Could not find a correct DN for host: %s. It will be ignored.'% host
+  return DNs
+
+
+def updateRabbitMQDatabase(newUsers, specialUsers = None):
+  """Updates the internal user database of RabbitMQ server.
+     The current user list from the database is compared to newUsers list.
+     The users that are not present in the database, but are in newUsers list
+     are added. The users that are present in the database, but are not in
+     newUsers list are deleted. The specialUser list contains logins that
+     will not be processed at all.
+  Args:
+    newUsers(list): user logins to be processed.
+    specialUsers(list): special users that will not be processed.
+  """
+  if specialUsers is None:
+    specialUsers = ['admin', 'dirac', 'ala', 'O=client,CN=kamyk']
```

Comment:
I will change it.

This module should be in the second PR so I will move it there.